src/client.cpp

```diff
@@ -473,7 +473,7 @@ void UpdatePort(unsigned int port, bool connected, const game_controller* contro
 
   const unsigned int device = CInputManager::Get().GetDevice(port);
 
-  if (CLIENT)
+  if (CLIENT && (int)port >= 0)
```

Comment:
UpdatePort() is exclusively for joysticks and shouldn't be called with a mouse port. If this isn't clear, we could rename to `UpdateJoystickPort()` or improve the documentation.